### PR TITLE
feat(bigquery): Add support for AllowNonIncrementalDefinition and Staleness on MaterializedView

### LINKED ISSUE #8668: bigquery: support max staleness and non-incremental materialized views
**Is your feature request related to a problem? Please describe.**
I can't create the materialized views I want via the Go SDK

**Describe the solution you'd like**

### Non-Incremental

From the docs, there is an api field `allow_non_incremental_definition`
https://cloud.google.com/bigquery/docs/materialized-views-create#create-non-inc

```json
{
  "kind": "bigquery#table",
  "tableReference": {
    "projectId": "my_project",
    "datasetId": "my_dataset",
    "tableId": "my_mv_table"
  },
  "materializedView": {
    "query": "`SELECT`
        s_store_sk,
        SUM(ss_net_paid) AS sum_sales,
        APPROX_QUANTILES(ss_net_paid, 2)[safe_offset(1)] median
      FROM my_project.my_dataset.store
      LEFT OUTER JOIN my_project.my_dataset.store_sales
        ON ss_store_sk = s_store_sk
      GROUP BY s_store_sk
      HAVING median < 40 OR median is NULL`",

    "allow_non_incremental_definition": true
  }
  "maxStaleness": "4:0:0"
}
```

Admittedly that field doesn't exist in the api docs yet, maybe because of the preview status
https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#materializedviewdefinition

### Max Staleness

https://cloud.google.com/bigquery/docs/materialized-views-create#create-max_staleness
```json
{
  "kind": "bigquery#table",
  "tableReference": {
    "projectId": "project-id",
    "datasetId": "my_dataset",
    "tableId": "my_mv_table"
  },
  "materializedView": {
    "query": "select product_id,sum(clicks) as
                sum_clicks from project-id.my_dataset.my_base_table
                group by 1"
  }
  "maxStaleness": "4:0:0"
}
```

max staleness does show up already in the api docs
https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#resource:-table

**Describe alternatives you've considered**
Using SQL, but that's much more difficult to do declaratively

### PR DESCRIPTION
**Pull request by alvarowolfx**
Fixes #8668

### CODE REVIEW

File: bigquery/table.go [hunk 1]
```diff
@@ -317,19 +317,29 @@ type MaterializedViewDefinition struct {
...
 	// at which this this materialized view will be refreshed.
 	RefreshInterval time.Duration
+
+	// AllowNonIncrementalDefinition for materialized view definition.
+	// The default value is false.
+	AllowNonIncrementalDefinition bool
+
+	// MaxStaleness of data that could be returned when materialized view
+	// is queried (formatted as Google SQL Interval type).
+	MaxStaleness string
```
Comment (shollyman) [inline bigquery/table.go:327]: Should we just use https://pkg.go.dev/cloud.google.com/go/bigquery#IntervalValue here and convert?

**Comment by derekperkins** [comment]
Is there anything preventing this from getting merged?

**Comment by derekperkins** [comment]
Thanks for adding this. Is there anything preventing this from getting merged?

**Comment by shollyman** [comment]
TODO: verify the behavior for maxStaleness at the table and within the MVD message.  We may want to simply expose the top level message instead.

**Comment by derekperkins** [comment]
Following up on this, hoping to get it in a release sooner than later

**Review by shollyman — APPROVED** [review]

**Comment by derekperkins** [comment]
Thanks!!

**Pull request merged**